kvision-tools/kvision-gradle-plugin/src/main/kotlin/io/kvision/gradle/KVisionPlugin.kt

```diff
@@ -121,12 +121,10 @@ abstract class KVisionPlugin @Inject constructor(
             dependsOn(tasks.all.browserProductionWebpack)
         }
 
-        afterEvaluate {
-            tasks.all.processResources.configureEach {
-                exclude("**/*.pot")
-                dependsOn(tasks.all.compileKotlinJs)
-                finalizedBy("convertPoToJson")
-            }
+        tasks.all.processResources.configureEach {
+            exclude("**/*.pot")
+            dependsOn(tasks.all.compileKotlinJs)
+            finalizedBy(convertPoToJsonTask)
```

Comment:
using `finalizedBy` means `convertPoToJsonTask` will *always* run, even if `processResources` fails. Normally you'd use `finalizedBy` for ensuring task output got cleaned, or if a test failed a report-task still created a report. I don't think that's the situation here, so it's not quite semantically right. But it's not a big problem, Because `KVConvertPoTask` has `@SkipWhenEmpty` it should automatically skip.

I'm not sure what the task graph is, so I can only make a guess at an alternative: I assume users don't directly run `processResources`, but instead run `assemble` (or something similar?). So if `assemble` depends on `processResouces`, then that should produce the required task ordering.

Reply:
The goal is to run `convertPoToJson` automatically after `processResources` is executed. The problem is the `processResources` task is executed by a bunch of different tasks (for development and production) and not a single task like `assemble`. So it's hard to set correct dependencies for all these tasks with `dependsOn`. At least I'm not sure how to do this.

Reply:
Ah okay, thanks. Well it looks like `finalizedBy` is the right thing to do 👍